src/bayfield_main.cpp

```diff
@@ -35,41 +38,140 @@ SDL_Surface *copy_frame(const char *executable_path) {
     return ret;
 }
 
+/* Show a simple error message box or print error to console if box fails */
+void show_simple_error(std::string title, std::string message) {
+    title.insert(0, "BAYFIELD: ");
+    if (SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_ERROR, title.c_str(), message.c_str(), nullptr)) {
+        // if message box could not be shown, last ditch console print
+        std::cerr << message << std::endl;
+    }
+    return;
+}
+
+/* Show a simple information box or print message to console if box fails */
+void show_simple_info(std::string title, std::string message) {
+    title.insert(0, "BAYFIELD: ");
+    if (SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_INFORMATION, title.c_str(), message.c_str(), nullptr)) {
+        // if message box could not be shown, last ditch console print
+        std::cout << message << std::endl;
+    }
+    return;
+}
+
 int main(int argc, char** args) {
+
+    int PROGRAM_SCALE = 3;
+    bool DRAW_FRAME = true;
+    int GPU_P = 0;
+
+    if (SDL_Init(SDL_INIT_VIDEO)) { // SDL could not initialize
+        show_simple_error("SDL INIT FAILURE", "Could not initialize required SDL subsystems.");
+        return 1;
+    }
+
     char *rom_path = NULL;
-    SDL_Init(SDL_INIT_VIDEO);
-
-    // Check if the argv provided path is a real file first. OS X will sometimes
-    // add args if it was a UI launch.
-    if (argc > 1) {
-        if (access(args[1], F_OK) == 0) {
-            rom_path = strdup(args[1]);
-        } else {
-            std::cerr << "The specified file does not exist." << std::endl;
+
+    // args parsing
+    for (int i = 1; i < argc; i++) {
+        if (args[i][0] == '-') {
+            switch (args[i][1]) {
+                case 'v': //version
+                    show_simple_info("VERSION INFO", 
+                                    "BAYFIELD EMULATOR " + VERSION_BFE() +
+                                    "\nBUILT WITH " + VERSION_CXX() +
+                                    "\nCOMPILED AGAINST SDL " + VERSION_SDL_COMPILED() +
+                                    "\nLINKED AGAINST SDL " + VERSION_SDL_LINKED());
+                    return 0;
+                case 's': //scale
+                    if (i + 1 < argc) {
+                        int val = atoi(args[i+1]);
+                        PROGRAM_SCALE = ((val < 1) ? PROGRAM_SCALE : val);
+                        i++;
+                    }
+                    break;
+                case 'p': //palette
+                    if (i + 1 < argc) {
+                        GPU_P = atoi(args[i+1]);
+                        i++;
+                    }
+                    break;
+                case 'f': //border
+                    DRAW_FRAME = false;
+                    break;
+            }
+        }
+        else if (args[i][0] == '?') {
```

Comment:
We might want to use a different flag, since ? is a shell globbing character it will lead to weird behaviour if there are any files with a single character name in the directory

Reply:
that's a fun feature to keep you on your toes

I'm just going to rework/remove the help thing, it's a messy solution and I don't like how it came out